### ISSUE TITLE
`missed-obfuscation` incorrectly counts receiver `ρ` as a void leading to spurious defect reports

### ISSUE THREAD

**Issue by gemshrine** [objectionary/lints#1461]
**What happens**

`missed-obfuscation` counts the receiver `ρ` among the void attributes a formation is told to obfuscate. A formation whose only void is its own receiver gets a defect it cannot act on, because `ρ` is the receiver and there is nothing to rename it to.

On lints `de102d68` with eo-parser `0.63.0`:

```eo
+package foo
+version 0.0.0

[^] > thing
  ^.other > @
```

```
void: name=[ρ]   (U+03C1)
DEFECT line 4: The formation "thing" must obfuscate its void attributes before decorating another object
```

A formation with a real void gets the same message, which is the intended case:

```eo
[x] > thing
  x > @
```

and one with no voids at all is silent, as it should be.

Over `eo-runtime/src/main/eo` (`objectionary/eo` at `8655f8755e`) the rule reports **706** defects, and in **92** of them the only void of the formation is `ρ`.

**Why**

The selector takes any void whose name is not prefixed `cant-` (`design/missed-obfuscation.xsl`):

```xml
<xsl:for-each select="//o[eo:abstract(.) and @name and o[@name='φ'] and o[@base='∅' and @name and not(starts-with(@name, 'cant-'))]]">
```

Writing the receiver down as `[^]` emits `<o base="∅" name="ρ"/>`, which matches. The motive talks only about attributes a caller provides:

> A formation that decorates another object should obfuscate its void attributes. Otherwise, an attribute of the formation may shadow an attribute with the same name in the decoratee

`ρ` shadows nothing of the kind: it is the receiver every formation has, and `eo-printer` writes it as `^` rather than as a name a reader could confuse with an attribute of the decoratee.

This is the same shape as #1299, where `many-void-attributes` counts `ρ` against its cap of five. The fix there is the fix here.

**How to fix**

Leave the receiver out of the selector:

```xml
<xsl:for-each select="//o[eo:abstract(.) and @name and o[@name='φ'] and o[@base='∅' and @name and @name != 'ρ' and not(starts-with(@name, 'cant-'))]]">
```

A pack for it: `[^] > thing` with `^.other > @` must give `defects: 0`, while `[^ x] > thing` must still be reported for `x`.

**Comment by gemshrine** [comment]
The mirror case is worth deciding at the same time: `no-attribute-formation` also counts `ρ` as a void, so it stays silent on `[^] > x`, even though from the author’s side that formation has no parameters either.

```
[] > x      -> no-attribute-formation: The formation "x" does not have any void attributes
[^] > x     -> (silent; only missed-obfuscation fires)
[q] > x     -> (silent for no-attribute-formation, correctly)
```

Whichever way `ρ` is decided, both selectors should agree on it.

**Comment by 0crat** [comment]
@gemshrine Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+3518](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
